src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -0,0 +1,169 @@
+                Logger.LogError("An error occurred while retrieving an access token: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                return OAuthTokenResponse.Failed(new Exception("An error occurred while retrieving an access token."));
+            }
+            using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync());
+            using var alipay_system_oauth_token_response = JsonDocument.Parse(payload.RootElement.GetProperty("alipay_system_oauth_token_response").ToString());
+
+            return OAuthTokenResponse.Success(alipay_system_oauth_token_response);
+        }
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(
+            [NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties,
+            [NotNull] OAuthTokenResponse tokens)
+        {
+            var sortedparams = new SortedDictionary<string, string>()
+            {
+                ["app_id"] = Options.ClientId,
+                ["method"] = "alipay.user.info.share",
+                ["format"] = "JSON",
+                ["charset"] = "UTF-8",
+                ["sign_type"] = "RSA2",
+                ["timestamp"] = Clock.UtcNow.ToString("yyyy-MM-dd HH:mm:ss"),
+                ["version"] = "1.0",
+                ["auth_token"] = tokens.AccessToken,
+            };
+            var signparames = RSA2Sign(sortedparams);
+            if (Options.NotValidateSign)
+            {
+                signparames.Remove("timestamp");
+                signparames.Remove("sign");
+            }
+            string address = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, signparames);
+
+            using var response = await Backchannel.GetAsync(address);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+            using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync());
+
+            using var alipay_user_info_share_response = JsonDocument.Parse(payload.RootElement.GetProperty("alipay_user_info_share_response").ToString());
```

Comment:
I think all of the code above this line is exactly the same as in the previous method. The commonalities can be factored into a private method to move the duplication.